safe_transaction_service/loggers/custom_logger.py

```diff
@@ -0,0 +1,250 @@
+class HttpResponseLog:
+    status: int
+    endTime: int
+    totalTime: int
+    errorMessage: str | None = None
+
+
+@dataclass
+class ErrorInfo:
+    function: str
+    line: int
+    exceptionInfo: str | None = None
+
+
+@dataclass
+class TaskInfo:
+    name: str
+    id: str
+    args: tuple | None = None
+    kwargs: dict | None = None
+
+
+@dataclass
+class ContextMessageLog:
+    session: str | None = None
+    httpRequest: HttpRequestLog | None = None
+    httpResponse: HttpResponseLog | None = None
+    errorInfo: ErrorInfo | None = None
+    taskInfo: TaskInfo | None = None
+    extraData: dict | None = None
+
+
+@dataclass
+class JsonLog:
+    level: str
+    timestamp: int
+    context: str
+    message: str
+    lineno: int
+    contextMessage: ContextMessageLog | None = None
+
+    def _remove_null_values_from_log(self, json_log: dict):
+        """
+        Delete keys with the value ``None`` in a dictionary, recursively.
+        """
+        for key, value in list(json_log.items()):
+            if value is None:
+                del json_log[key]
+            elif isinstance(value, dict):
+                self._remove_null_values_from_log(value)
+        return json_log
+
+    def to_json(self):
+        return json.dumps(self._remove_null_values_from_log(asdict(self)))
+
+
+def get_milliseconds_now():
+    return int(time.time() * 1000)
+
+
+def http_request_log(
+    request, timestamp: int | None = None, log_data: bool = False
+) -> HttpRequestLog:
+    """
+    Generate httpRequestLog from provided request
+    """
+    route = request.resolver_match.route if request.resolver_match else request.path
+    return HttpRequestLog(
+        url=str(route),
+        urlReplaced=request.path,
+        method=request.method,
+        timestamp=timestamp or get_milliseconds_now(),
+        body=request.data if log_data else None,
+    )
+
+
+class SafeJsonFormatter(logging.Formatter):
+    """
+    Json formatter with following schema
+    {
+        level: str,
+        timestamp: Datetime,
+        context: str,
+        message: str,
+        contextMessage: <contextMessage>
+    }
+    """
+
+    def format(self, record) -> str:
+        """
+        Format logging record as json string.
+        """
+
+        if record.levelname == "ERROR":
+            exception_info: str | None = None
+            # Check if the error contains exception data
+            if record.exc_info:
+                exc_type, exc_value, exc_tb = record.exc_info
+                exception_info = "".join(
+                    traceback.format_exception(exc_type, exc_value, exc_tb)
+                )
+
+            record.error_detail = ErrorInfo(
+                function=record.funcName,
+                line=record.lineno,
+                exceptionInfo=exception_info,
+            )
+
+        # Generate context_message
+        context_message = ContextMessageLog(
+            session=record.session if hasattr(record, "session") else None,
```

Comment:
```suggestion
            session=getattr(record, 'session',None),
```

And same with the others